db/db_iter_stress_test.cc

```diff
@@ -97,7 +97,8 @@ struct StressTestIterator : public InternalIterator {
 
   bool MaybeFail() {
     if (rnd->Next() >=
-        std::numeric_limits<uint64_t>::max() * error_probability) {
```

Comment:
Example doing essentially the same thing without casts:

https://github.com/facebook/rocksdb/blob/bc51e33d9cd507697b74dae6f68ee6cf42e632cb/cache/cache_bench.cc#L174-L175

Note that the left side of >= is still implicitly converted (with loss of precision), so there's a risk of compiler warning whack-a-mole.

Reply:
You can forget about this concern at the moment. When I have time, I'll put together something that makes it easier to write code like this.